src/ui/components/AnswerOption/AnswerOption.test.tsx

```diff
@@ -0,0 +1,47 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import { describe, it, expect, vi } from 'vitest';
+
+import AnswerOption from './AnswerOption';
+
+import type { IProps } from './types/IProps';
+
+describe('AnswerOption Component', () => {
+  const baseProps: IProps = {
+    label: 'Female',
+    value: 'female',
+    selected: false,
+    onChange: vi.fn(),
+  };
+
+  it('renders the label', () => {
+    render(<AnswerOption {...baseProps} />);
+    expect(screen.getByText('Female')).toBeInTheDocument();
+  });
+
+  it('shows as not selected by default', () => {
+    render(<AnswerOption {...baseProps} />);
+    expect(screen.getByRole('radio')).not.toBeChecked();
+  });
+
+  it('shows selected state when selected is true', () => {
+    render(<AnswerOption {...baseProps} selected />);
+    expect(screen.getByRole('radio')).toBeChecked();
+  });
+
+  it('calls onChange with value when clicked', () => {
+    const handleChange = vi.fn();
+    render(<AnswerOption {...baseProps} onChange={handleChange} />);
+    fireEvent.click(screen.getByLabelText('Female'));
```

Comment:
The test performs fireEvent.click(screen.getByLabelText('Female')). If the input is hidden (display:none), getByLabelText may not find the input (it searches accessible elements by default). This makes the click and handler assertion flaky or fail.

**Sugerencia:**
```typescript
Either click the visible label element by querying text (e.g. fireEvent.click(screen.getByText('Female'))) or ensure the input is visually hidden in an accessible way so getByLabelText returns the input. Using userEvent from '@testing-library/user-event' is also recommended for better simulation: await userEvent.click(screen.getByText('Female'));
```